docs(hipblaslt): add PR prose-style guide to pr-quality skill (#10819)
ROCm/rocm-libraries: projects/hipblaslt/skills/hipblaslt-pr-quality/prose-style-context.md

# Prose and Style Context

## Purpose

Use this context when writing pull-request descriptions, review comments, handoffs, design notes, and engineering summaries for any change in rocm-libraries, whether the code is C++ or Python and whether it touches production code, build files, or tests.

The writing must help a busy engineer understand the change without first reading the code or earlier pull requests. Technical accuracy is required, but repository knowledge is not assumed.

## Reader model

Write for a high-school senior who:

- has written novice Python and C++;
- understands functions, classes, lists, dictionaries, pointers, files, exceptions, tests, and command-line tools;
- has not seen rocm-libraries or the changed code;
- may not know GPU terminology, ROCm conventions, math-library or kernel concepts, acronyms, or processes; and
- will stop reading if the explanation is vague, inflated, or unnecessarily long.

> An experienced engineer should still find the text precise. Plain language must not remove important technical facts.

Reviewer: David, I really like the goal here. Forcing the description to stand alone for someone who does not already know the component is the right instinct, and the "explain before you name it" part is especially useful.

Where I got hung up is the high-school-senior framing. On my ear it pushes the prose a bit toward elementary, even when the material is technical. I still want the accessibility you are aiming for; I just think we can aim that at a technically literate engineer instead.

```suggestion
## Reader model

Write for a technically literate engineer at roughly an undergraduate level who:

- understands common programming concepts in Python or C++;
- has not worked in rocm-libraries or the changed component;
- may not know GPU terminology, ROCm conventions, math-library concepts, project-specific acronyms, or internal processes;
- expects unfamiliar terms to be explained when they matter to the change; and
- values concrete evidence and a coherent technical argument over either jargon or oversimplification.

Use the tone and sentence structure of well-edited technical reporting in a major newspaper: accessible, precise, and natural. An experienced engineer should still find the description useful, while an engineer new to the domain should be able to follow it without opening the diff.
```

## Priority order

When goals conflict, use this order:

1. Correctness
2. Clarity
3. Necessary context
4. Concision
5. Elegance

Do not shorten a sentence by replacing an explanation with jargon.

## Length limits

- Aim for about 200 words for a pull-request description.
- Use 201–350 words when definitions materially improve understanding.
- Never exceed 300 words without explicit approval.
- Prefer sentences under 25 words. Treat 30 words as a warning.
- Keep most paragraphs to two through four sentences.

> These are readability limits, not targets. A complete 160-word description is better than a padded 200-word description.

Reviewer: I like that you put correctness and clarity ahead of elegance, and I think the length guidance is coming from a good place. Short, reviewable descriptions are a real improvement over the sprawling ones we sometimes write.

The hard cap is the part I would soften. When I tried this on one of my own PRs, getting under 300 words meant dropping exact run IDs and live workflow results, which were the strongest evidence in the description. Your priority order already says necessary context beats concision; I would just make the wording match that.

```suggestion
## Length guidance

- Aim for about 200 words when that is enough to explain the change.
- Use additional length when it preserves necessary context, test evidence, risk analysis, or an unusual review decision.
- Remove repetition, discarded development history, and generic process language before removing technical evidence.
- Prefer sentences under 30 words, but vary sentence length naturally.
- Keep most paragraphs to two through four sentences unless a longer paragraph develops one connected technical argument.

These are readability signals, not acceptance criteria. Correctness, clarity, necessary context, and evidence take precedence over length. A focused 400-word description with reproducible test results is better than a 200-word description that omits how the change was verified.
```

## Sentence-level gate

Apply every question below to every sentence. If any answer is no, rewrite the sentence.

1. Can the reader identify what object, function, file, or process the sentence is about?

> 2. If a name or acronym appears for the first time, is it defined in plain language?
> 3. Are the remaining terms common to a Python programmer? If not, are they explained?
> 4. Does the sentence make sense without reading an earlier pull request?
> 5. Does it express one main idea with a clear subject and verb?
> 6. Is it concrete about an input, output, behavior, failure, or consequence?
> 7. Does the reader know why this fact matters?
> 8. Could a simpler verb replace an abstract noun phrase?
> 9. Does it avoid slogans, sales language, and clever phrasing?
> 10. Does it describe the current net change rather than discarded development history?
> 11. Is every factual claim supported by the diff, a test result, or an authoritative source?
> 12. Could the target reader paraphrase it correctly after one reading?
> 
> ## Explain concepts before identifiers

Reviewer: The sentence-level gate is one of my favorite parts of this PR. The checks about naming the object, defining terms, and grounding claims in evidence are excellent.

I would just add a paragraph-level companion so we do not accidentally encourage a very flat cadence. "One main idea per sentence" is good advice, but if every sentence follows the same short pattern, the prose starts to feel choppy even when the content is solid.

```suggestion
12. Could the target reader paraphrase it correctly after one reading?

## Sentence cadence

Apply these checks to each paragraph after checking its individual sentences:

- Vary sentence length naturally. Short sentences should emphasize an important result, not become the default rhythm.
- Combine closely related facts when the relationship between them is clearer in one sentence.
- Use transitions to show cause, contrast, qualification, and consequence.
- Let a paragraph develop one connected idea rather than reading like a list with the bullets removed.
- Read the paragraph aloud. If every sentence has the same length or structure, revise the cadence without removing technical detail.

## Explain concepts before identifiers
```
